africanus/calibration/tests/test_utils.py

```diff
@@ -26,65 +21,83 @@ def give_lm(n_dir):
     return lm
 
 
-def give_flux(n_dir):
-    return 0.1 + np.random.random(n_dir)
+def give_flux(n_dir, n_chan, cor_shape, alpha, freq, freq0):
+    w = freq/freq0
+    flux = np.zeros((n_dir, n_chan) + cor_shape, dtype=np.float64)
+    for d in range(n_dir):
+        tmp_flux = np.random.random(size=cor_shape)
+        for v in range(n_chan):
+            flux[d, v] = tmp_flux * w[v]**alpha
+    return flux
+
 
+corr_shape_parametrization = pytest.mark.parametrize(
+    'corr_shape, jones_shape',
+    [((2,), (2,)),  # DIAG_DIAG
+     ((2, 2), (2,)),  # DIAG
+     ((2, 2), (2, 2)),  # FULL
+     ])
 
-def make_dual_pol_data(sigma_n, n_dir, sigma_f):
+
+def make_data(sigma_n, sigma_f, n_time, n_chan, n_ant,
```

Comment:
This could be a pytest factory fixture http://doc.pytest.org/en/latest/fixture.html living in a conftest.py. See for example the [xarray-ms ms fixture](https://github.com/ska-sa/xarray-ms/blob/master/xarrayms/tests/conftest.py)

```python
@pytest.fixture
def calibration_factory():
  def impl(sigma_n, sigma_f, n_time, n_chan, n_ant, ...):
      data_dict = {}
      # Fill in data_dict
      ...
      return data_dict

  return impl
```

def test_calibration(calibration_facttory):
   data_dict = calibration_factory(sigma_n, sigma_f, n_time, ...)
```
```

Reply:
Alright, I had a go at this but not sure if I did it correctly. Maybe you can have a look?